### DISCUSSION TITLE
How to paint or fill with transparent color

### DISCUSSION THREAD

**Discussion by Toninoso** [tannerhelland/PhotoDemon#538] [Q&A] [answered]
I have Windows BMP file that contains transparent sections. There is one piece that is currently white and I would like to make it transparent. I used color picker to pick transparent pixel but when I use picked color it fills the area with black color. 
I haven't been able to paint or fill with 'colors' that have opacity 0. What am I missing? Thanks!

**Comment by tannerhelland** [comment, marked as answer]
Hi @Toninoso .  To erase pixels (make them transparent), you can use the `eraser tool` (keyboard shortcut `E`).

**Reply by Toninoso** [reply]
Thanks!. Though, for filling the area it is a bit clumsy working pixel by pixel. But it works.

**Reply by tannerhelland** [reply]
You're very welcome!

If you're using a selection tool to select the region you want to make transparent (like the magic wand tool), you could then use the `Edit > Clear` menu to erase the entire selected region - maybe that's an easier approach?

**Reply by tannerhelland** [reply]
(I forgot to mention that if a selection is active, you can also simply press the `Delete` key to erase the selected region - i.e. to force the whole region to opacity 0.  This is identical to using the `Edit > Clear` menu.)

**Reply by Toninoso** [reply]
Nice, thanks again!

**Comment by manfromarce** [comment]
Being able to use transparent / semi-transparent colors in the brush and paint bucket tools would be a nice addition in my opinion too. If I remember correctly Paint.Net allows to set opacity (0-255) in the color wheel while PhotoDemon does not, so this is why the eyedropper picks black color. However I don't think the BMP format supports transparency (?).

**Reply by tannerhelland** [reply]
Hi @manfromarce .  In PhotoDemon, the paintbrush, pencil, and paint bucket tools all support transparent colors.  Like Photoshop and GIMP, opacity is set via *tool settings* instead of in the color wheel.  For example, here is a screenshot from the paint bucket tool:

![image](https://github.com/tannerhelland/PhotoDemon/assets/1930029/77d1ec6e-b265-47d1-9589-bbbcc9d6eca3)

There is an identical opacity setting in the paintbrush and pencil tools.

One new option I could add is an "overwrite" or "replace" blend mode (different software calls these different names).  In this blend mode in Paint.NET, for example, if you paint with blue at 50% opacity over a white background, it will first *erase* the white background, then *replace* the erased pixels with blue at 50% opacity.  This can provide a way to "paint" semi-transparent (or transparent) pixels directly into an image, instead of "blending" them with existing pixels.

**Reply by Toninoso** [reply]
@manfromarce  Historically that is true that BMP format didn't support transparency. For a while one had a new DIB format that did. Nowadays it is often all merged under files with BMP extension and smart tools know how to parse the header. PhotoDemon correctly shows checkered pattern when I load a file with BMP extension that has transparency.

**Reply by manfromarce** [reply]
@tannerhelland Thank you for the information, I had missed the opacity option for some reason.
Adding a "Replace" blend mode would be a good usability improvement indeed.
